stellargraph/layer/graph_classification.py

```diff
@@ -0,0 +1,236 @@
+                node features (size batch_size x N x F),
+                graph adjacency matrix (batch_size x N x N),
+                where N is the number of nodes in the graph, and F is the dimensionality of node features.
+
+        Returns:
+            Keras Tensor that represents the output of the layer.
+        """
+        features, A = inputs
+
+        h_graph = K.batch_dot(A, features)
+        output = K.dot(h_graph, self.kernel)
+
+        # Add optional bias & apply activation
+        if self.bias is not None:
+            output += self.bias
+        output = self.activation(output)
+
+        return output
+
+
+@experimental(reason="Missing unit tests")
+class GraphClassification:
+    """
+    A stack of :class:`GraphClassificationConvolution` layers together with a Keras `GlobalAveragePooling1D` layer
+    that implement a supervised graph classification network using the GCN convolution operator
+    (https://arxiv.org/abs/1609.02907).
+
+    The model minimally requires specification of the GCN layer sizes as a list of ints
+    corresponding to the feature dimensions for each hidden layer,
+    activation functions for each hidden layers, and a generator object.
+
+    To use this class as a Keras model, the features and pre-processed adjacency matrix
+    should be supplied using the :class:`GraphGenerator` class.
+
+    Notes:
+      - The inputs are tensors provided by the :class:`GraphGenerator` object.
+
+    Examples:
+        Creating a graph classification model from a list of :class:`StellarGraph`
+        objects (``graphs``). We also add two fully connected dense layers using the last one for binary classification
+        with `softmax` activation::
+
+            generator = GraphGenerator(graphs)
+            model = GraphClassification(
+                             layer_sizes=[32, 32],
+                             activations=["elu","elu"],
+                             generator=generator,
+                             dropout=0.5
+                )
+            x_inp, x_out = model.build()
+            predictions = Dense(units=8, activation='relu')(x_out)
+            predictions = Dense(units=2, activation='softmax')(predictions)
+
+    Args:
+        layer_sizes (list of int): list of output sizes of the graph GCN layers in the stack.
+        activations (list of str): list of activations applied to each GCN layer's output.
+        generator (GraphGenerator): an instance of :class:`GraphGenerator` class constructed on the graphs used for
+            training.
+        bias (bool, optional): toggles an optional bias in graph convolutional layers.
+        dropout (float, optional): dropout rate applied to input features of each GCN layer.
+        kernel_regularizer (str): normalization applied to the GCN layer kernels.
+    """
+
+    def __init__(
+        self, layer_sizes, activations, generator, bias=True, dropout=0.0, **kwargs
```

Comment:
Instead of adding new examples for #801, maybe this could list out the arguments that are computed in `_get_regularisers_from_keywords`?

Reply:
I've updated with regards to kwargs but I want to keep the example because this model cannot be used unless at least another layer is added to its output, e.g., a Dense layer. For other models such as GCN, the last convolutional layer can also be the classification layer if it has the correct activation applied but not for the graph classification model.

Reply:
> I want to keep the example because this model cannot be used unless at least another layer is added to its output, e.g., a Dense layer

I'm not sure how that's relevant to #801? Maybe I was confusing, I meant: "Instead of adding a new class that has the problem noted in #801, maybe this could ...".

Reply:
I thought you meant the example of how to use the model not the list for #801. All good,  I'll just close this thread.